include/data_struct.h

```diff
@@ -186,9 +200,15 @@ class BaseLSTMStates {
     std::vector<float> mu_ha, var_ha, mu_f_ga, var_f_ga, jcb_f_ga, mu_i_ga,
         var_i_ga, jcb_i_ga, mu_c_ga, var_c_ga, jcb_c_ga, mu_o_ga, var_o_ga,
         jcb_o_ga, mu_ca, var_ca, jcb_ca, mu_c, var_c, cov_i_c, cov_o_tanh_c;
+
     std::vector<float> mu_c_prev, var_c_prev, mu_h_prev, var_h_prev, mu_h_prior,
         var_h_prior, mu_c_prior, var_c_prior;
 
+    std::vector<std::vector<float>> mu_h_priors, var_h_priors, mu_c_priors,
```

Comment:
I would avoid 2D vector as much as possible. See my comments within SLinear and SLSTM

Reply:
Done